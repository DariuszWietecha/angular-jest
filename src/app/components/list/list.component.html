<div class="text-right" *ngIf="showCreateButton === true">
	<a [routerLink]="['/', createEndpoint, 'create']" class="btn btn-primary" >Create</a>
</div>

<h4>{{ title }}</h4>
<div class="d-flex flex-wrap">
  <div *ngFor="let item of items" class="card col-4">
    <template [ngTemplateOutletContext]='{item: item}' [ngTemplateOutlet]="templateVariable"></template>
  </div>
</div>