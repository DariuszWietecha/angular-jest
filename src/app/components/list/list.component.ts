import { Component, Input, ContentChild, TemplateRef } from '@angular/core';

@Component({
  selector: 'app-list',
  templateUrl: './list.component.html',
  styleUrls: ['./list.component.scss']
})
export class ListComponent {
  @Input() createEndpoint = '';
  @Input() items: any[];
  @Input() showCreateButton = true;
  @Input() title: string;
  @ContentChild(TemplateRef, {static: true}) templateVariable: TemplateRef<any>;

}
