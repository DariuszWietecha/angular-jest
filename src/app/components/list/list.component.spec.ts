import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { ListComponent } from './list.component';

describe('ListComponent', () => {
  let component: ListComponent;
  let fixture: ComponentFixture<ListComponent>;
  const categoriesListMock = require('../../../../test/data/categories/categories-get.json');


  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ListComponent],
      imports: [RouterTestingModule],
    })
      .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ListComponent);
    component = fixture.componentInstance;
    component.items = categoriesListMock;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('snapShot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
