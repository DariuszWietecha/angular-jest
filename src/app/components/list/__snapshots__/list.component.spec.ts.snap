// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListComponent snapShot 1`] = `
<app-list
  createEndpoint=""
  items={[Function Array]}
  showCreateButton={[Function Boolean]}
  templateVariable="undefined"
>
  <div
    class="text-right"
  >
    <a
      class="btn btn-primary"
      href="//create"
      ng-reflect-router-link="/,,create"
    >
      Create
    </a>
  </div><h4>
    
  </h4><div
    class="d-flex flex-wrap"
  >
    
    <div
      class="card col-4"
    >
      <template
        ng-reflect-ng-template-outlet-context="[object Object]"
      />
    </div>
    <div
      class="card col-4"
    >
      <template
        ng-reflect-ng-template-outlet-context="[object Object]"
      />
    </div>
  </div>
</app-list>
`;
